x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplication.java

```diff
@@ -99,7 +110,6 @@ public SearchApplication(StreamInput in) throws IOException {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeString(name);
-        out.writeStringArray(indices);
         out.writeOptionalString(analyticsCollectionName);
```

Comment:
Similar change here:

```suggestion
if (out.getTransportVersion().before(INDICES_REMOVED_TRANSPORT_VERSION) {
        out.writeStringArray(indices); // old behaviour. New behaviour does not serialize indices, so no need to do anything else
}
        out.writeOptionalString(analyticsCollectionName);
```